backend/app/services/summary_service.py

```
@@ -0,0 +1,519 @@
+        summary_data = document.summary or {}
+        summary_data["status"] = "generating"
+        summary_data["updated_at"] = datetime.now().isoformat()
+        document.summary = summary_data
+        flag_modified(document, "summary")
+        self.db.commit()
+
+        try:
+            # 5. Get document content
+            content = await self._get_document_content(document)
+            if not content:
+                raise Exception("Failed to get document content")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)